### ISSUE TITLE
Add carbon count

### ISSUE THREAD

**Issue by emilienschultz** [emilienschultz/pyactivetigger#86]
Use https://codecarbon.io/ to compute an estimate

+ time spent for computation

**Comment by emilienschultz** [comment]
De @eollion  : deux types d’approche pour calculer l’usage du modèle, mais aussi le Power Usage Efficiency (tout le cout d’existence de ton data center – y compris la lumiere – hors cout d’entrainement, et c’est environ de 1.8 pour Jean Zay).

Il y a deux approches :
online tools (Green algorithms, ML CO2 impact), MLCA,
or python packages (Carbon Tracker, Cumulator)
+ un capteur à mettre sur le GPU

Les outils en ligne sont très faciles à condition de connaître les ressources utilisées, mais à la bonne volonté des clients. le package python est un tout petit peu plus compliqué (faut des droits), le capteur demande l'acces root complet

Le package python peut être plus précis, ca marche bien en local. Mais c’est plus difficile sur les infrastructures partagées pour accéder au niveau root.

De manière générale, tout est de l’ordre de l’estimation un peu pifometre.

Sinon, une autre option serait de mettre, au niveau des GPU, ce genre de calculateur (AI Power Meter, voir Git et poster conférence ICT4S), qui donne une consommation plus précise.

**Comment by eollion** [comment]
Re-upping this one in the light of the past and coming controversies about the use of AI. One strong point of Tigger is that it is a low-energy consumption (frugal) device. If we had
- a counter for each job/session
- a counter for each person
it would be a great specification.

That the counter is precise or not is not a fundamental issue at this stage (we document the calculation, we thus let people decide if it is serious). And once we have a counter in place, we could update it later. But having an instrument could be useful.

**Comment by eollion** [comment]
I agree with all of the above, but I feel that having a counter would force people to think about it. Can't we just give info about the number of minutes used on a GPU for the bert part? And for the generate, if you click on the "carbon counter", it would tell you "you know that this is not parcimonious, dont you"?

**Comment by jboelaert** [comment]
Agreed

**Comment by eollion** [comment]
Re-upping this issue, as I think we should have a carbon counter in Tigger in the near future. 
This is an important issue, in several papers we say this is important to know this, and Tigger is eco-friendly.

**Comment by emilienschultz** [comment]
I added a minimal solution in the **monitor** panel for each task

- the time x memory used
- codecarbon  (codecarbon won't work on our dev instance in Onyxia)

For the moment, I don't add it in the user side, we need to discuss what we want do display, how to interpret it and the limit if this information
